dashboard/app/models/user.rb

```diff
@@ -1116,7 +1110,7 @@ def next_unpassed_progression_level(script)
     # the script in multiple places (i.e. via level swapping) there's some potential
     # for strange behavior.
     sl_level_ids = script.script_levels.map(&:level_ids).flatten
-    ul_with_sl = user_levels_by_level(script).select do |level_id, _ul|
+    ul_with_sl = Queries::UserProgress.user_levels_by_level(self, script).select do |level_id, _ul|
```

Comment:
likewise, this could probably just be
```suggestion
    ul_with_sl = user_levels.where(script: script, level_id: sl_level_ids)
```